libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -960,7 +1021,7 @@ impl NodeAPI for Greenlight {
                 "send_pay route to pay: {:?}, received_amount = {}",
                 route, to_pay_msat
             );
-
+            self.wait_channel_reestablished(&max.path).await?;
```

Comment:
As far as I know greenlight has a mechanism to wait for channel re-establishment before attempting to pay, what is the benefit of adding such mechanism on the client too?

Reply:
Or perhaps this mechanism doesn't exist in `send_pay`  and that's why it is added here?

Reply:
> Or perhaps this mechanism doesn't exist in `send_pay` and that's why it is added here?

exactly: https://github.com/Blockstream/greenlight/issues/522